### ISSUE TITLE
property is undefined after mapping

### ISSUE THREAD

**Issue by maxime-aubry** [nartc/mapper#595]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

After is called the mapper, i get a date as a string, as i wanted.
But 'id' property is undefined.
I must add 'forMember' function for 'id' property too.

### Models/DTOs/VMs

``` TypeScript
import { AutoMap } from '@automapper/classes';

export class TransformationAuditGetAllResponseState {
  @AutoMap()
  id!: string;

  @AutoMap()
  transformationDate!: string;
}

export class TransformationAuditGetAllResponseDto {
  @AutoMap()
  id!: string;

  @AutoMap()
  transformationDate!: Date;
}
```

### Mapping configuration

``` TypeScript
export const mapper: Mapper = createMapper({
  strategyInitializer: classes(),
});

createMap(
  mapper,
    TransformationAuditGetAllResponseDto,
    TransformationAuditGetAllResponseState,
  forMember(
      (dest: TransformationAuditGetAllResponseState) => dest.transformationDate,
    mapFrom((source: TransformationAuditGetAllResponseDto) => moment(source.transformationDate).format('DD/MM/YYYY hh:mm:ss A'))
  ),
);
```

### Expected behavior

Value of 'id' property of TransformationAuditGetAllResponseDto should be mapped for 'id' property of TransformationAuditGetAllResponseState.

### Package

- [ ] I don't know.
- [X] <code>@automapper/core</code>
- [X] <code>@automapper/classes</code>
- [ ] <code>@automapper/nestjs</code>
- [ ] <code>@automapper/pojos</code>
- [ ] <code>@automapper/mikro</code>
- [ ] <code>@automapper/sequelize</code>
- [ ] Other (see below)

### AutoMapper version

8.8.1

### Additional context

typescript version : 5.3.3

**Comment by maxime-aubry** [comment]
Mapper works if i add type on properties.

for example :

``` Typescript
import { AutoMap } from '@automapper/classes';

export class TransformationAuditGetAllResponseState {
  @AutoMap(() => String)
  id!: string;

  @AutoMap(() => String)
  transformationDate!: string;
}

export class TransformationAuditGetAllResponseDto {
  @AutoMap(() => String)
  id!: string;

  @AutoMap(() => Date)
  transformationDate!: Date;
```

Is there a way to do with another way ?

**Comment by bradws** [comment]
Yes I'm getting the same result as you. However it only seems to occur in my testing environment (I use [vitest](https://vitest.dev/)). If I run it as normal transpiled javascript, it is fine and I don't need to specify the `()=>String`

**Comment by geek-sajjad** [comment]
Can I start working on this bug? @maxime-aubry @bradws

**Comment by nartc** [comment]
Sorry for the very late response. I do not see a direct fix for this undefined-property mapping report in the current 9.0 migration on branch `chore/9.0-migrate`, so I am closing it as stale/not planned.

**Comment by nartc** [comment]
Reopening this. The earlier bulk triage closure was too aggressive here.

Reason: the undefined-property mapping report is recent enough to keep open for follow-up rather than close under the 9.0 umbrella.

Keeping this open for explicit follow-up instead of treating it as closed by the 9.0 migration work.

**Comment by nartc** [comment]
v9 fixes this. check the integration test https://github.com/nartc/mapper/blob/main/packages/integration-test/src/classes/issues/595/595.spec.ts
